src/EntityFramework/Migrations/Design/CSharpMigrationCodeGenerator.cs

```diff
@@ -1536,7 +1536,14 @@ protected virtual string TranslateColumnType(PrimitiveTypeKind primitiveTypeKind
         /// <returns> The quoted identifier. </returns>
         protected virtual string Quote(string identifier)
         {
-            return "\"" + identifier + "\"";
+            using (var writer = new StringWriter())
+            {
+                using (var provider = CodeDom.Compiler.CodeDomProvider.CreateProvider("CSharp"))
+                {
+                    provider.GenerateCodeFromExpression(new CodeDom.CodePrimitiveExpression(identifier), writer, null);
+                    return writer.ToString();
+                }
+            }
```

Comment:
I much prefer you're original suggestion to using CodeDOM. The following is probably sufficient. (It's what we use in EF Core)
```C#
return "\"" + identifier.Replace("\\", "\\\\").Replace("\"", "\\\"") + "\"";
```